src/snapshot/SnapshotServer.cpp

```diff
@@ -136,75 +136,22 @@ SnapshotServer::recvPushSnapshotDiffs(const uint8_t* buffer, size_t bufferSize)
           ->localLock();
     }
 
-    // Iterate through the chunks passed in the request
+    // Convert chunks to snapshot diff objects
+    std::vector<SnapshotDiff> diffs;
     for (const auto* chunk : *r->chunks()) {
-        uint8_t* dest = snap->data + chunk->offset();
```

Comment:
The following logic around merging snapshot diffs has also been moved to `SnapshotData`.